services/motion/builtin/move_request.go

```diff
@@ -0,0 +1,179 @@
+	goal := spatialmath.GeoPointToPose(destination, origin)
+
+	// convert GeoObstacles into GeometriesInFrame with respect to the base's starting point
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles, origin)
+
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	worldState, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// construct limits
+	straightlineDistance := goal.Point().Norm()
+	if straightlineDistance > maxTravelDistanceMM {
+		return nil, fmt.Errorf("cannot move more than %d kilometers", int(maxTravelDistanceMM*1e-6))
+	}
+	limits := []referenceframe.Limit{
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+	ms.logger.Debugf("base limits: %v", limits)
+
+	if extra != nil {
+		if profile, ok := extra["motion_profile"]; ok {
+			motionProfile, ok := profile.(string)
+			if !ok {
+				return nil, errors.New("could not interpret motion_profile field as string")
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+	}
+
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return nil, resource.NewNotFoundError(componentName)
+	}
+	b, ok := baseComponent.(base.Base)
+	if !ok {
+		return nil, fmt.Errorf("cannot move component of type %T because it is not a Base", baseComponent)
+	}
+
+	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, localizer, limits, kinematicsOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// create a new empty framesystem which we add the kinematic base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+	kbf := kb.Kinematics()
+	if err := fs.AddFrame(kbf, fs.World()); err != nil {
+		return nil, err
+	}
+
+	// TODO(RSDK-3407): this does not adequately account for geometries right now since it is a transformation after the fact.
+	// This is probably acceptable for the time being, but long term the construction of the frame system for the kinematic base should
+	// be moved under the purview of the kinematic base wrapper instead of being done here.
+	offsetFrame, err := referenceframe.NewStaticFrame("offset", movementSensorToBase.Pose())
+	if err != nil {
+		return nil, err
+	}
+	if err := fs.AddFrame(offsetFrame, kbf); err != nil {
+		return nil, err
+	}
+
+	return &moveRequest{
+		planRequest: &motionplan.PlanRequest{
+			Logger:             ms.logger,
+			Goal:               referenceframe.NewPoseInFrame(referenceframe.World, goal),
+			Frame:              offsetFrame,
+			FrameSystem:        fs,
+			StartConfiguration: referenceframe.StartPositions(fs),
+			WorldState:         worldState,
+			Options:            extra,
+		},
+		actuator: kb,
+		execute: 
```

Comment:
where do we define how execute works?
i think this is missing

Reply:
oops